flink-runtime/src/main/java/org/apache/flink/runtime/security/token/DefaultDelegationTokenManager.java

```diff
@@ -118,10 +133,13 @@ private Map<String, DelegationTokenProvider> loadProviders() {
                             provider.serviceName());
                 }
             } catch (Exception | NoClassDefFoundError e) {
-                LOG.warn(
+                // The intentional general rule is that if a provider's init method throws exception
+                // then stop the workload
+                LOG.error(
                         "Failed to initialize delegation token provider {}",
                         provider.serviceName(),
                         e);
+                throw new RuntimeException(e);
```

Comment:
Shouldn't we do either logging or throwing the exception and if we decide to throw the exception, adding the log message as the error message might be useful. :thinking: `FlinkRuntimeException` might be useful since it indicates that the exception was actually initiated by Flink code. ...just as an idea.

Reply:
Makes sense to use `FlinkRuntimeException`, changed on both places.